lib/screens/profile/password_bloc.dart

```diff
@@ -0,0 +1,49 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:openflutterecommerce/repos/password_repository.dart';
+import 'package:openflutterecommerce/screens/profile/password_event.dart';
+import 'package:openflutterecommerce/screens/profile/password_state.dart';
+
+class PasswordBloc extends Bloc<PasswordEvent, PasswordState> {
+  final PasswordRepository passwordRepository;
+
+  PasswordBloc({@required this.passwordRepository})
+      : assert(passwordRepository != null);
+
+  @override
+  PasswordState get initialState => PasswordInitialState();
+
+  @override
+  Stream<PasswordState> mapEventToState(PasswordEvent event) async* {
+    if (event is ChangePasswordEvent) {
+      if (event.currentPassword.length == 0) {
```

Comment:
could you replace for "currentPassword.isEmpty" the other validations too.

Reply:
Please I don't understand what you mean. could you elaborate?

Reply:
Fixed it